src/json.hpp

```diff
@@ -2640,7 +2582,7 @@ class lexer
             const auto x = std::strtoull(yytext.data(), &endptr, 10);
 
             // we checked the number format before
-            assert(endptr == yytext.data() + yylen);
+            assert(endptr == yytext.data() + yytext.size());
```

Comment:
Could `yytext.end()` be used here?

Reply:
Yes, but we're testing the output of a function that was originally passed yytext.data(), so this is probably clearer.